chia/data_layer/data_store.py

```diff
@@ -631,17 +650,15 @@ async def autoinsert(
         tree_id: bytes32,
         hint_keys_values: Optional[Dict[bytes, bytes]] = None,
         use_optimized: bool = True,
-        *,
-        lock: bool = True,
     ) -> bytes32:
-        async with self.db_wrapper.locked_transaction(lock=lock):
-            was_empty = await self.table_is_empty(tree_id=tree_id, lock=False)
+        async with self.db_wrapper.write_db():
```

Comment:
if you don't save the connection object here, why are you acquiring a write lock for the DB?